devito/data/allocators.py

```diff
@@ -306,6 +307,55 @@ def put_local(self):
         return self._node == 'local'
 
 
+class ExternalAllocator(MemoryAllocator):
+
+    """
+    External allocator for definig Function symbols with data loaded by the
+    user, without copying it to Devito's allocated memory.
+
+    Parameters
+    ----------
+    numpy_array : numpy.array
+        Array of data.
+
+    Returns
+    -------
+    pointer, None
+        The first element of the tuple is the reference that can be used to
+        access the data.
+
+    Notes
+    -------
+    * Use ExternalAllocator and pass a reference to the external memory when
+      creating a function. This Function will now use this memory as it's f.data.
+
+    * If the data present in this external memory is valuable, provide a noop
+      initialiser, or else Devito will reset it to 0.
+
+    Example
+    --------
+    >>> shape = (2, 2)
+    >>> space_order = 0
+    >>> numpy_array = np.ones(shape, dtype=np.float32)
+    >>> g = Grid(shape)
+    >>> f = Function(name='f', space_order=space_order, grid=g,
+    ...      allocator=ExternalAllocator(numpy_array), initializer=lambda x: None)
```

Comment:
at this point you could add another couple of lines

```
>>> f.data[0, 1] = 2
>>> numpy_array
[[ 1 2 ]
 [ 1 1 ]]
```

and could also add this file to `test_docstrings.py`

Reply:
oki